muzzle/src/main/java/io/opentelemetry/javaagent/tooling/HelperInjector.java

```diff
@@ -146,18 +149,29 @@ public static void setHelperInjectorListener(HelperInjectorListener listener) {
     helperInjectorListener = listener;
   }
 
-  private Map<String, byte[]> getHelperMap() throws IOException {
+  private Map<String, Supplier<byte[]>> getHelperMap() {
     if (dynamicTypeMap.isEmpty()) {
-      Map<String, byte[]> classnameToBytes = new LinkedHashMap<>();
-
-      ClassFileLocator locator = ClassFileLocator.ForClassLoader.of(helpersSource);
+      Map<String, Supplier<byte[]>> result = new LinkedHashMap<>();
 
       for (String helperClassName : helperClassNames) {
-        byte[] classBytes = locator.locate(helperClassName).resolve();
-        classnameToBytes.put(helperClassName, classBytes);
+        result.put(
+            helperClassName,
+            () -> {
+              try {
+                return StreamDrainer.DEFAULT.drain(
+                    helpersSource.getResourceAsStream(
+                        helperClassName.replace('.', '/') + ".class"));
```

Comment:
is this just inlining `ClassFileLocator.ForClassLoader.of(helpersSource).locate(helperClassName).resolve()`?

Reply:
Yes, I changed it use `ClassFileLocator`.